SimplenoteIntents/IntentHandler.swift

```diff
@@ -51,13 +43,45 @@ extension IntentHandler: NoteWidgetIntentHandling {
     }
 
     func defaultNote(for intent: NoteWidgetIntent) -> WidgetNote? {
-        guard let dataController = try? WidgetDataController(coreDataManager: coreDataManager) else {
+        guard let note = try? WidgetDataController(context: coreDataManager.managedObjectContext).firstNote() else {
             return nil
         }
 
-        guard let note = try? dataController.firstNote() else {
-            return nil
-        }
         return WidgetNote(identifier: note.simperiumKey, display: note.limitedTitle)
     }
 }
+
+extension IntentHandler: ListWidgetIntentHandling {
+    func provideTagOptionsCollection(for intent: ListWidgetIntent, with completion: @escaping (INObjectCollection<WidgetTag>?, Error?) -> Void) {
+        guard let dataController = try? WidgetDataController(context: coreDataManager.managedObjectContext) else {
+            completion(nil, WidgetError.appConfigurationError)
+            return
+        }
+
+        guard let tags = dataController.tags() else {
+            completion(nil, WidgetError.fetchError)
+            return
+        }
+
+        // Return collection to intents
+        let collection = tagNoteInObjectCollection(from: tags)
+        completion(collection, nil)
+    }
+
+    private func tagNoteInObjectCollection(from tags: [Tag]) -> INObjectCollection<WidgetTag> {
+        let allNotesWidgetTag: [WidgetTag] = [WidgetTag(identifier: SimplenoteConstants.allNotesTagIdentifier, display: Constants.allNotesDisplay)]
+        let fetchedWidgetTags = tags.map({ tag in
+            WidgetTag(identifier: tag.name ?? Constants.unnamedTag, display: tag.name ?? Constants.unnamedTag)
+        })
+        return INObjectCollection(items: allNotesWidgetTag + fetchedWidgetTags)
```

Comment:
Potentially bad idea / untested: Since the `List.intentdefinitions` don't support Enums with Associated Values, you **could** try something like this:

```
struct ListFilter {
    let kind: ListFilterKind
    let payload: String?
}

enum ListFilterKind {
   case allNotes
   case tag
}
```

The point is to "emulate" an associated values. When `kind == .tag` then `payload` is the tag name`. Otherwise, all notes.

Notes:
- This would allow you get rid of `TagsFilter`, and have a single structure
- Yes, this is a workaround
- Didn't test it, so not 100% on spot if this would work, or if there's any other fields required.
- This would also fix the `"All-Notes"` reserved keyword problem (having a Tag with that name would break the scheme)

WDYT? (Just brainstorming, I'm not 100% sold on the idea)